awx/main/tests/unit/utils/test_schedule_fast_forward.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+import datetime
+
+from django.utils.timezone import now
+
+from awx.main.models.schedules import fast_forward_date
+from dateutil.rrule import rrule, HOURLY, MINUTELY, MONTHLY
+
+
+def test_fast_forward_date_all_hours():
+    '''
+    Assert that the resulting fast forwarded date is included in the original rrule
+    occurrence list
+    '''
+    for interval in range(1, 24):
+        dtstart = now() - datetime.timedelta(days=30)
+        rule = rrule(freq=HOURLY, interval=interval, dtstart=dtstart)
+        new_datetime = fast_forward_date(rule)
+
+        # get occurrences of the rrule
+        gen = rule.xafter(new_datetime - datetime.timedelta(days=1), count=100)
+        found_matching_date = False
+        for occurrence in gen:
+            if occurrence == new_datetime:
+                found_matching_date = True
+                break
+
+        assert found_matching_date
+
+
+@pytest.mark.parametrize(
+    'freq, interval',
+    [
+        (MINUTELY, 15),
+        (MINUTELY, 120),
+        (MINUTELY, 60 * 24 * 3),
+        (HOURLY, 7),
+        (HOURLY, 24 * 3),
+    ],
```

Comment:
(cosmetic changes)
```suggestion
    ('freq', 'interval'),
    (
        pytest.param(MINUTELY, 15, id='every-15-minutes-minutely'),
        pytest.param(MINUTELY, 120, id='every-2-hours-minutely'),
        pytest.param(MINUTELY, 60 * 24 * 3, id='every-3-days-minutely'),
        pytest.param(HOURLY, 7, id='every-7-hours-hourly'),
        pytest.param(HOURLY, 24 * 3, id='every-3-days-hourly'),
    ),
```

Reply:
incorporated this pattern in to my tests, although this specific test went away